lib/regexp/regexp.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package regexp provides regular expression related functions.
+package regexp // import "go.starlark.net/lib/regexp"
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+// Module regexp is a Starlark module of regular expression related functions.
+// The module defines the following functions:
+//
+//     compile(pattern) - Compile a regular expression pattern into a regular expression object, which
```

Comment:
"Compiles" (present not imperative)

"compiles a pattern in RE2 syntax (https://github.com/google/re2/wiki/Syntax) to a value of type 'regexp'"
(avoids 'object', and duplication of "regular expression", and states the type of the value)

Should we describe the equivalence relation as:
"Each call to compile returns a distinct regexp value." ?
(The other choices are: specify equivalence based on pattern strings, or leave it unspecified. My preference is the wording above.)

"matches, find, find_all, and other methods"
(parens f() indicate a call to a function named f)